src/hotspot/share/classfile/placeholders.hpp

```diff
@@ -134,16 +99,19 @@ class PlaceholderEntry : public HashtableEntry<Symbol*, mtClass> {
   bool remove_seen_thread(JavaThread* thread, PlaceholderTable::classloadAction action);
 
  public:
-  // Simple accessors, used only by SystemDictionary
-  Symbol*            klassname()           const { return literal(); }
-
-  ClassLoaderData*   loader_data()         const { return _loader_data; }
-  void               set_loader_data(ClassLoaderData* loader_data) { _loader_data = loader_data; }
+  PlaceholderEntry() :
+     _supername(nullptr), _definer(nullptr), _instanceKlass(nullptr),
+     _superThreadQ(nullptr), _loadInstanceThreadQ(nullptr), _defineThreadQ(nullptr) { }
 
   Symbol*            supername()           const { return _supername; }
   void               set_supername(Symbol* supername) {
+    if (_supername != nullptr) _supername->decrement_refcount();
     _supername = supername;
-    if (_supername != NULL) _supername->increment_refcount();
+    if (_supername != nullptr) _supername->increment_refcount();
+  }
```

Comment:
Could you use maybe_increment_refcount() instead of checking for NULL in the above code?

Reply:
Yes, I fixed it to use the maybe version instead.